xilem/src/view/progress_bar.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use masonry::widget;
+use xilem_core::{Mut, ViewMarker};
+
+use crate::{MessageResult, Pod, View, ViewCtx, ViewId};
+
+pub fn progress_bar(progress: Option<f64>) -> ProgressBar {
+    ProgressBar { progress }
+}
+
+pub struct ProgressBar {
+    progress: Option<f64>,
+}
+
+impl ViewMarker for ProgressBar {}
+impl<State, Action> View<State, Action, ViewCtx> for ProgressBar {
+    type Element = Pod<widget::ProgressBar>;
+    type ViewState = ();
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        ctx.with_leaf_action_widget(|_| Pod::new(masonry::widget::ProgressBar::new(self.progress)))
```

Comment:
Why is this an action widget? The progressbar widget doesn't seem to send actions as far as I can see